tests/unit/replicator_tests.py

```diff
@@ -210,6 +211,47 @@ def test_create_replication(self):
             ])
         )
 
+    def test_timeout_in_create_replication(self):
```

Comment:
I don't think it is necessary given that we'd just be testing underlying Requests functionality. One test of connect and one test of read would be sufficient for me that the pass-through of args is working as we desire.